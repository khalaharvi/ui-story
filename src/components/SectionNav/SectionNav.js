import React, { Children } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { createClassNames } from 'utils';

const cx = createClassNames('section-nav');

const SectionNav = ({ children, className }) => (
  <nav className={classnames(cx(''), className)}>
    <ul className={cx('list')}>
      {
        Children.map(children, child => (
          <li className={cx('list-item')}>{child}</li>
        ))
      }
    </ul>
  </nav>
);

SectionNav.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default SectionNav;
