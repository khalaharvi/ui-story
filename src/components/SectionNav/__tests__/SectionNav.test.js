import React from 'react';
import { shallow } from 'enzyme';

import SectionNav from '../SectionNav';

describe('<SectionNav />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SectionNav>
        <a href="/">Item 1</a>
        <a href="/">Item 2</a>
        <a href="/">Item 3</a>
      </SectionNav>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom className', () => {
    const wrapper = shallow(
      <SectionNav className="custom-section-nav">
        <a href="/">Item 1</a>
        <a href="/">Item 2</a>
        <a href="/">Item 3</a>
      </SectionNav>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
