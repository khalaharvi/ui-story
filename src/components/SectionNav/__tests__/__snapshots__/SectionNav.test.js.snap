// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionNav /> renders correctly 1`] = `
<nav
  className="dh-section-nav"
>
  <ul
    className="dh-section-nav-list"
  >
    <li
      className="dh-section-nav-list-item"
      key=".0"
    >
      <a
        href="/"
      >
        Item 1
      </a>
    </li>
    <li
      className="dh-section-nav-list-item"
      key=".1"
    >
      <a
        href="/"
      >
        Item 2
      </a>
    </li>
    <li
      className="dh-section-nav-list-item"
      key=".2"
    >
      <a
        href="/"
      >
        Item 3
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<SectionNav /> renders correctly with custom className 1`] = `
<nav
  className="dh-section-nav custom-section-nav"
>
  <ul
    className="dh-section-nav-list"
  >
    <li
      className="dh-section-nav-list-item"
      key=".0"
    >
      <a
        href="/"
      >
        Item 1
      </a>
    </li>
    <li
      className="dh-section-nav-list-item"
      key=".1"
    >
      <a
        href="/"
      >
        Item 2
      </a>
    </li>
    <li
      className="dh-section-nav-list-item"
      key=".2"
    >
      <a
        href="/"
      >
        Item 3
      </a>
    </li>
  </ul>
</nav>
`;
